lib/medtronic600/medtronic600Simulator.js

```diff
@@ -510,13 +382,122 @@ class Medtronic600Simulator {
     });
   }
 
+  applySuspendedBasals() {
```

Comment:
I didn't mean to imply your code here is complicated or unmaintainable :) I realize that this part of the simulator is inherently complex, so I don't think further refactoring will help. I just wanted to note that I didn't inspect the code in a whole lot of detail as I don't think it will be all that useful, and that a combination of unit tests and manual test strategies will do a better job of making sure it works. 

Do you think it will be possible for Eric to generate some of the edge cases and test scenarios that you discovered on Caty's pump, so that we can check the box that we manually verified them too?